analog_forecast.py

```diff
@@ -0,0 +1,297 @@
+        ),
+        required=True
+    )
+    parser.add_argument(
+        "-d", dest="ref_date", type=str, help="Date to use for analog search", required=True
+    )
+    parser.add_argument(
+        "-s",
+        dest="spatial_domain",
+        type=str,
+        help=f"Spatial domain to use for analog search and forecast. Options are {', '.join(list(luts.spatial_domains.keys()))}",
+        default="alaska",
+    )
+    parser.add_argument(
+        "-w", dest="workers", type=int, help="Number of workers to use for dask", default=8
+    )
+    args = parser.parse_args()
+    
+    return args.varname, args.ref_date, args.spatial_domain, args.workers
+
+
+def spatial_subset(da, bbox):
+    """Subset a DataArray spatially. Allows using a bbox that crosses the antimeridian. 
+    Trying to reassing the longitude coordinate before subsetting was causing intense slowdowns with xarray for some reason. So this function actually divides a bbox into two chunks, loads the data for each, and then combines back into a single dataset on the [0, 360) scale. 
+    
+    Args:
+        da (xarray.DataArray): dataset with longitude coordinate variable on [-180, 180) scale
+        bbox (tuple): bounding box for subdomain selection where 0 <= bbox[0] < bbox[2] < 360
+    
+    Returns:
+        sub_da (xarray.DataArray): dataset with longitude coordinate variable on [0, 360) scale
+    """
+    varname = da.name
+    if 0 <= bbox[0] < bbox[2] < 360:
+        da_w = da.sel(latitude=slice(bbox[3], bbox[1]), longitude=slice(-180, (bbox[2] - 360))).load()
+        da_e = da.sel(latitude=slice(bbox[3], bbox[1]), longitude=slice(bbox[0], 180)).load()
+        da_w = da_w.assign_coords(longitude=da_w.longitude.values + 360)
+        sub_da = xr.combine_by_coords([da_w, da_e])[varname]
+    else:  
+        sub_da = da.sel(latitude=slice(bbox[3], bbox[1]), longitude=slice(bbox[0], bbox[2]))
+    
+    return sub_da
+
+
+def read_subset_era5(spatial_domain, data_dir, varname, use_anom):
+    """Opens a connection to the ERA5 dataset for the variable of interest and subsets to the specified spatial region.
+    
+    Args:
+        spatial_domain (str): name of the spatial domain to subset to
+        data_dir (pathlib.PosixPath): path to directory containing ERA5 data
+        varname (str): name of the variable being used
+        use_anom (bool): whether or not to use anomalies
+        
+    Returns:
+        sub_da (xarray.DataArray): data array of ERA5 data subset to the spatial domain of interest
+    """
+    # get the bbox
+    bbox = luts.spatial_domains[spatial_domain]["bbox"]
+
+    # get filepath to data file used
+    # These will be for forecasts only so we want the real data, not anomaly
+    # open connection to file
+    if use_anom:
+        fp_lu_key = "anom_filename"
+    else:
+        fp_lu_key = "filename"
+    fp = data_dir.joinpath(luts.varnames_lu[varname][fp_lu_key])
+
+    # susbet the data spatially as was done for analogs, then extract
```

Comment:
The comments here don't seem to match the code below...there is no 14 day time slice happening in this section